### ISSUE TITLE
[data grid] Popper is not overflowing footer

### ISSUE THREAD

**Issue by luispresuelVenafi** [mui/mui-x#11708]
### Steps to reproduce

I'm basically experimenting the same exact issue as described in here:
https://stackoverflow.com/questions/77606791/how-to-make-popper-be-always-above-react-mui-datagrid-and-out-of-its-boundaries

Quoting the issue:
> I have a MUI DataGrid Pro table. In one column a cell can contain a tooltip. When tooltip's popper is shown, it is cut by the boundaries of a table and scrollbars appear:

![image](https://github.com/mui/mui-x/assets/77068025/1f654e43-3854-4b82-b4ef-7976f42a2834)

### For my case

If  tooltip is within the rows, there's no issue:

<img width="559" alt="issue-datagrid01" src="https://github.com/mui/mui-x/assets/77068025/86c62c97-d38f-40a3-bc68-fc2440e6408a">

however, nothing is able to get render above the footer if the tooltip text if is from the last row:

<img width="499" alt="issue-datagrid02" src="https://github.com/mui/mui-x/assets/77068025/15c17a25-0c59-4cca-ba1b-3f03191bf570">

Any workaround for this issue?

### Current behavior

Popper is render behind of z-inder of datagrid footer

### Expected behavior

Popper should be able to render on top of datagrid footer

### Your environment

    "@mui/material": "5.14.7",
    "@mui/x-data-grid-pro": "6.12.0",

using chrome, but issue also in any other browser.

**Search keywords**: popper datagrid tooltip

**Comment by michelengelen** [comment]
Hey @luispresuelVenafi did you try using the `disablePortal` prop?

```
<Tooltip
  ...
  PopperProps={{disablePortal: true}}
  ...
/>
```

**Comment by michelengelen** [comment]
oh, and could you please provide a minimal working example? That would help a lot! 🙇🏼

**Comment by luispresuelVenafi** [comment]
@michelengelen Yeah, already provided that value.

Also try to hack my way adding z-index and important like the following:

```
renderCell: (params: any) => {
        return (
          (params.row.metadata as GcpCertificateMetadata)?.name && (
            <>
              <Tooltip
                sx={{ '& .MuiTooltip-popper': { zIndex: '99999999999 !important', position: 'absolute !important' } }}
                PopperProps={{
                  disablePortal: true,
                  style: { zIndex: '99999999999 !important' },
                  modifiers: [
                    {
                      name: 'preventOverflow',
                      enabled: false,
                    },
                  ],
                }}
                title={t(my-title')}
                arrow
                open={openedTooltipId === params.row.id}
                placement="bottom"
              >
                <Box>
                  <IconButton
                    size="small"
                    role="button"
                    aria-label={'my-aria-label'}
                    onClick={() => doCopyfunction()}
                  >
                    <ContentCopyIcon fontSize="small" }} />
                  </IconButton>
                </Box>
              </Tooltip>
            </>
          )
        );
```

**Comment by luispresuelVenafi** [comment]
@michelengelen Yeah, already provided that value and didn't help. I actually I needed to provided, otherwise the Tooltip wouldn't be visible in the first place.

Also try to hack my way adding z-index and important like the following:

```
renderCell: (params: any) => {
        return (
          (condition && (
            <>
              <Tooltip
                sx={{ '& .MuiTooltip-popper': { zIndex: '99999999999 !important', position: 'absolute !important' } }}
                PopperProps={{
                  disablePortal: true,
                  style: { zIndex: '99999999999 !important' },
                  modifiers: [
                    {
                      name: 'preventOverflow',
                      enabled: false,
                    },
                  ],
                }}
                title={t(my-title')}
                arrow
                open={condition}
                placement="bottom"
              >
                <Box>
                  <IconButton
                    size="small"
                    role="button"
                    aria-label={'my-aria-label'}
                    onClick={() => doCopyfunction()}
                  >
                    <ContentCopyIcon fontSize="small" }} />
                  </IconButton>
                </Box>
              </Tooltip>
            </>
          )
        );

```

but didn't work

**Comment by luispresuelVenafi** [comment]
@michelengelen Here's a codesandbox I was able to reproduce the issue:

https://codesandbox.io/p/sandbox/datagrid-tooltip-is-not-overflowing-footer-on-last-column-rcc52f

We are using DataGridPro if that helps on this issue be solved.

**Comment by michelengelen** [comment]
> @michelengelen Here's a codesandbox I was able to reproduce the issue:
> 
> https://codesandbox.io/p/sandbox/datagrid-tooltip-is-not-overflowing-footer-on-last-column-rcc52f
> 
> Just over over the "copy" Icon and you will see the Tooltip will render properly, but the last one in the row, won't be able to overflow the Datagrid.
> 
> We are using DataGridPro. Didn't know what's the Order ID or Support key, or if there was a process to get them.

OK, it was actually easier than I though. Disabling the portal was actually preventing that. My bad. When you enable the portal again it works:

<img width="770" alt="Screenshot 2024-01-18 at 09 26 24" src="https://github.com/mui/mui-x/assets/32863416/399befc6-ca11-48f3-bab9-3136f18d9a5b">

Here is the updated [sandbox](https://codesandbox.io/p/sandbox/datagrid-tooltip-is-not-overflowing-footer-on-last-column-forked-vjphrl?file=%2Fsrc%2FApp.tsx%3A61%2C18)

**Comment by luispresuelVenafi** [comment]
@michelengelen so if I remove:
`disablePortal: true,`
Our tooltip won't render at all

**Comment by michelengelen** [comment]
> @michelengelen so if I remove: `disablePortal: true,` Our tooltip won't render at all

That indeed indicates a different problem. 

> I don't see anything more relevant in our case rather than the datagrid is inside a drawer. I may need to improve to sample codesandbox to me more like our case

Ok, when you have a sandbox we can view that more accurately shows your specific problem we can help you even better.

In the meantime I will try to replicate a situation with a grid inside a drawer!

**Comment by luispresuelVenafi** [comment]
Hi there @michelengelen 

I created this codesandbox that mimics on a higher level how the component live (the opened tooltips were done on purpose for testing the rendering):

https://codesandbox.io/p/sandbox/datagrid-in-drawer-tooltip-working-overflow-tjfmcj?file=%2Fsrc%2FApp.tsx%3A107%2C29

Unfortunately I was not able to reproduce the issue, but we found a workaround that is somewhat enough for us for now. I'll close the issue as the symptoms are mostly showing that issue is coming from our end and not the library.

We'll re-open if we ever manage to reproduce in a codesandbox environment or maybe we come back if we the proper fix.

Thank you very much for your active response
